advisor/src/main/kotlin/advisors/OssIndex.kt

```diff
@@ -64,11 +64,18 @@ class OssIndex(name: String, config: OssIndexConfiguration) : AdviceProvider(nam
 
     private val service by lazy {
         OssIndexService.create(
-            url = config.serverUrl,
-            client = OkHttpClientHelper.buildClient()
+            config.serverUrl,
+            config.username,
+            config.password,
+            OkHttpClientHelper.buildClient()
         )
     }
 
+    private val getComponentReport by lazy {
```

Comment:
What do you think about explicitly specifying the type of `getComponentReport()` for readability? 

Alternatively, this performance optimization can be dropped in favor of a (IMO more readable) function. To me it sacrifices readability without an obvious need.

What do you think?

Reply:
> What do you think?

I think it's not less readable than a wrapping function, and (letting the performance aspect aside) on the contrary more elegant.

Adding an explicit type would actually *reduce* readability, as this is a function reference to a `suspend` function, so the type would be something like `KSuspendFunction1<ComponentReportRequest, List<ComponentReport>>`.

Finally, using a function would require to pass `config` as an argument, as functions do not have access to constructor arguments like property initialization has.

So all in all, I'd prefer to keep it as-is.

*Edit:* Here the direct comparison

```
private val getComponentReport by lazy {
    val hasCredentials = config.username != null && config.password != null
    if (hasCredentials) service::getAuthorizedComponentReport else service::getComponentReport
}
```

vs

```
private suspend fun getComponentReport(config: OssIndexConfiguration, components: ComponentReportRequest): List<ComponentReport> {
    val hasCredentials = config.username != null && config.password != null
    return if (hasCredentials) service.getAuthorizedComponentReport(components) else service.getComponentReport(components)
}
```

to me the former is less repetitive, more elegant, and thus more readable and better.

Reply:
I had something like below in mind, however - let's just leave it as is.

```
private suspend fun ComponentReportRequest.execute(): List<ComponentReport> =
   when {
        config.hasCredentials() -> service.getAuthorizedComponentReport(this)
        else -> service.getComponentReport(this)
   }

private fun OssIndexConfiguration.hasCredentials() = config.username != null && config.password != null
```

Reply:
I like the extension function, though. Let me add that in a follow-up PR.